src/lib/accessibility/FocusZone/FocusZone.types.ts

```diff
@@ -0,0 +1,174 @@
+   */
+  direction?: FocusZoneDirection
+
+  /**
+   * Optionally provide a selector for identifying the intial active element.
+   */
+  defaultActiveElement?: string
+
+  /**
+   * If set, the FocusZone will not be tabbable and keyboard navigation will be disabled.
+   * This does not affect disabled attribute of any child.
+   */
+  disabled?: boolean
+
+  /**
+   * Element type the root element will use. Default is "div".
+   */
+  elementType?: keyof React.ReactHTML
+
+  /**
+   * If set, will cycle to the beginning of the targets once the user navigates to the
+   * next target while at the end, and to the end when navigate to the previous at the beginning.
+   */
+  isCircularNavigation?: boolean
+
+  /**
+   * If provided, this callback will be executed on keypresses to determine if the user
+   * intends to navigate into the inner zone. Returning true will ask the first inner zone to
+   * set focus.
+   */
+  isInnerZoneKeystroke?: (ev: React.KeyboardEvent<HTMLElement>) => boolean
+
+  /**
+   * Sets the aria-labelledby attribute.
+   */
+  ariaLabelledBy?: string
+
+  /**
+   * Sets the aria-describedby attribute.
+   */
+  ariaDescribedBy?: string
+
+  /**
+   * Callback for when one of immediate children elements gets active by getting focused
+   * or by having one of its respective children elements focused.
+   */
+  onActiveElementChanged?: (element?: HTMLElement, ev?: React.FocusEvent<HTMLElement>) => void
+
+  /**
+   * Deprecated at v1.12.1. DIV props provided to the FocusZone will be mixed into the root element.
+   * @deprecated
+   */
+  rootProps?: React.HTMLAttributes<HTMLDivElement>
+
+  /**
+   * Callback method for determining if focus should indeed be set on the given element.
+   * @param {HTMLElement} element The child element within the zone to focus.
+   * @returns True if focus should be set to the given element, false to avoid setting focus.
+   */
+  onBeforeFocus?: (childElement?: HTMLElement) => boolean
+
+  /** Allow focus to move to root */
+  allowFocusRoot?: boolean
+
+  /**
+   * Allows tab key to be handled to tab through a list of items in the focus zone,
+   * an unfortunate side effect is that users will not be able to tab out of the focus zone
+   * and have to hit escape or some other key.
+   * @deprecated Use 'handleTabKey' instead.
+   *
+   */
+  allowTabKey?: boolean
+
+  /**
+   * Allows tab key to be handled to tab through a list of items in the focus zone,
+   * an unfortunate side effect is that users will not be able to tab out of the focus zone
+   * and have to hit escape or some other key.
+   */
+  handleTabKey?: FocusZoneTabbableElements
+
+  /**
+   * A callback method to determine if the input element should lose focus on arrow keys
+   *  @param {HTMLInputElement} inputElement The input element which is to loose focus.
+   *  @returns True if input element should loose focus or false otherwise.
+   */
+  shouldInputLoseFocusOnArrowKey?: (inputElement: HTMLInputElement) => boolean
```

Comment:
Hey hey!  A callback that returns a boolean named `should*` 👍 

Not required but it would be helpful to note why this is useful in the doc.  Trying to imagine a use case for this...